From bc527da9f85072cc0e8ae5c552427eafc0909882 Mon Sep 17 00:00:00 2001
From: Louis-Alexis Eyraud <louisalexis.eyraud@collabora.com>
Date: Thu, 10 Apr 2025 12:34:00 +0200
Subject: [PATCH 50/79] arm64: dts: mt8365-evk: Enable mt6357 vibr and vsim2
 regulators by default

On the Mediatek Genio 350-EVK board, two power supplies are needed to
power on either the Ethernet PHY (Realtek RTL8201F) or HDMI Bridge IC
(ITE IT66121FN):
 - VIBR_PMU (mt6357_vibr_reg)
 - VSIM2_PMU (mt6357_vsim2_reg)

They are declared as regulators in mt6357.dtsi include and used in
the board devicetree but only as power supplies for the ITE IT66121FN
(it66121_hdmi node) and not for the Ethernet (ethernet node).

On this board, the Ethernet and HDMI output are two functionialities
that cannot exist simultaneously and are selected with a DIP switch.
So when the IT66121FN device is disabled by configuration, these two
power supplies are not powered on when they should be and the Ethernet
is not fully functional.

So to cover these various configuration cases,  add in mt8365-evk.dts
the 'regulator-always-on' property for both regulators with the
following rationale:
- The VSIM2_PMU (1.8v line) is used as a general pull-up so it seems
assumed it's always on by the hardware designer.
- The VIBR_PMU (3.3v line) is used as a vcc for various switches that
are configured via a jumper on the board as well. So it seems it was
intended to be always on too.

Signed-off-by: Louis-Alexis Eyraud <louisalexis.eyraud@collabora.com>
---
 arch/arm64/boot/dts/mediatek/mt8365-evk.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt8365-evk.dts b/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
index b5dd5ef9fa11..bd7bf23da183 100644
--- a/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
@@ -406,11 +406,19 @@ &mt6357_pmic {
 	mediatek,micbias1-microvolt = <1700000>;
 };
 
+&mt6357_vibr_reg {
+	regulator-always-on;
+};
+
 &mt6357_vsim1_reg {
 	regulator-min-microvolt = <1800000>;
 	regulator-max-microvolt = <1800000>;
 };
 
+&mt6357_vsim2_reg {
+	regulator-always-on;
+};
+
 &pio {
 	aud_default_pins: audiodefault-pins {
 		clk-dat-pins {
-- 
2.43.0

